alpakaConfig.cmake

```diff
@@ -373,6 +373,8 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
             OPTION(ALPAKA_CUDA_FTZ "Set flush to zero for GPU" OFF)
             OPTION(ALPAKA_CUDA_SHOW_REGISTER "Show kernel registers and create PTX" OFF)
             OPTION(ALPAKA_CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps (folder: nvcc_tmp)" OFF)
+            OPTION(ALPAKA_CUDA_NVCC_EXPT_EXTENDED_LAMBDA "Enable experimental, extended host-device lambdas in NVCC" ON)
+            OPTION(ALPAKA_CUDA_NVCC_EXPT_RELAXED_CONSTEXPR "Enable experimental, relaxed constexpr in NVCC" ON)
```

Comment:
I investigated the following with NVCC 8.0.44 on Hypnos (HZDR):

Alpaka `dev` *tests* so far build :heavy_check_mark: / fail to build :heavy_multiplication_x: with:
- :heavy_check_mark: : :heavy_plus_sign: `extended lambda` and :heavy_plus_sign: `relaxed constexpr`
- :heavy_check_mark: : :heavy_plus_sign: `extended lambda` and :heavy_minus_sign: `relaxed constexpr`

- :grey_question: : :  :heavy_minus_sign: `extended lambda` and :heavy_plus_sign: `relaxed constexpr`
- :grey_question: :  :heavy_minus_sign: `extended lambda` and :heavy_minus_sign: `relaxed constexpr`

:grey_question: is in progress